src/transformers/models/blip_2/convert_blip_2_original_to_pytorch.py

```diff
@@ -155,6 +155,8 @@ def convert_blip2_checkpoint(
     else:
         tokenizer = AutoTokenizer.from_pretrained("google/flan-t5-xl")
 
+    tokenizer.model_input_names = ["input_ids", "attention_mask"]
```

Comment:
This is pretty hacky. It should at the very least only be done in the BertTokenizer branch to make explicit why this is needed (including an explanatory comment)

Reply:
I don't see what's hacky about this, `model_input_names` is a genuine attribute of `PretrainedTokenizer` that has been there since BERT/GPT-2 came out (it's not used a lot). As can be seen [here](https://github.com/huggingface/transformers/blob/51296712900f28023cecea6ff7333af4b3c81d30/src/transformers/tokenization_utils_base.py#L1461-L1462), a tokenizer will return `token_type_ids` if it's present in the `model_input_names`.

A less "hacky" way would be to pass `model_input_names` directly in the `from_pretrained` method, which the Transformers library also supports.

Reply:
What I'm trying to say is that one can avoid lines line [these](https://github.com/huggingface/transformers/blob/51296712900f28023cecea6ff7333af4b3c81d30/src/transformers/models/blip_2/processing_blip_2.py#L59) by appropriately setting the `model_input_names` attribute of the tokenizer, which is something we can still do for the `Blip2ImageTextRetrieval` models on the hub as they are just added and not yet in a stable release.

Reply:
Ah, yes, I see. Although both modifying `model_input_names` and `return_token_ids` are somewhat hacky: they modify a class attribute which might be depended upon for other behaviours / assumed to have certain properties. Passing in `model_input_names` to the init is better, as this is more likely to correctly propagate any changes to any other dependant attributes / variables.